### ISSUE TITLE
can't delete Claim compositionRevisionSelector label

### ISSUE THREAD

**Issue by mateusz-lubanski-sinch** [crossplane/crossplane#3992]
### What happened?

I labeled my composition with 3 labels (`semverFull`, `semverMinor`, `semverPatch`)
I added `semverFull` compositionRevisionSelector label on my Claim
I wanted replace compositionRevisionSelector label on my claim from `semverFull` to `semverPatch`
It turned out that I can't delete previous label

### How can we reproduce it?

We added 3 labels to our Composition:

```
kubectl get compositions mysqlinstances.rds.aws.crossplane.sinch.com-v1alpha1 -o json | jq ".metadata.labels"
{
  "semverFull": "v1.22.7",
  "semverMinor": "v1.x.x",
  "semverPatch": "v1.22.x"
}
```

Than I selected on my claim `semverFull` label via `compositionRevisionSelector`

```
apiVersion: rds.aws.crossplane.sinch.com/v1alpha1
kind: MySQLInstance
metadata:
  name: mysql-1
spec:
  compositionRevisionSelector:
    matchLabels:
      semverFull: "v1.22.6"
```

```
kubectl apply -f mysql-1.yaml
mysqlinstance.rds.aws.crossplane.sinch.com/mysql-1 created
```

Than I wanted to replace `semverFull` label with `semverPatch`

```
apiVersion: rds.aws.crossplane.sinch.com/v1alpha1
kind: MySQLInstance
metadata:
  name: mysql-1
spec:
  compositionRevisionSelector:
    matchLabels:
      semverPatch: "v1.22.x"
```
```
kubectl apply -f mysql-1.yaml
mysqlinstance.rds.aws.crossplane.sinch.com/mysql-1 configured
```

After applying above changes I run below command to remove previous label
```
kubectl patch mysqlinstances.rds.aws.crossplane.sinch.com mysql-1 --type json -p='[{"op": "remove", "path": "/spec/compositionRevisionSelector/matchLabels/semverFull"}]'
```

After checking claim it turned out that it is **not** possible to delete previous label selector on `mysql-1`
```
kubectl get mysqlinstances mysql-1 -o yaml | yq ".spec.compositionRevisionSelector"
matchLabels:
  semverFull: v1.22.7
  semverPatch: v1.22.x
```

### What environment did it happen in?
EKS 1.22
Crossplane 1.11.3

**Comment by mateusz-lubanski-sinch** [comment]
Same behaviour applies when:

1. In CompositeResourceDefinition I specify parameters.engineVersion property as mandatory
2. I create Claim and set parameters.engineVersion
3. I changed CompositeResourceDefinition `parameters.engineVersion` property to optional (I set default value in Composition instead)
4. I wanted delete from Claim `parameters.engineVersion` parameter so value defined in Composition should be used
After removing above propertycrossplane adds it again to my Claim
